internal/graph/graph.go

```diff
@@ -431,12 +433,22 @@ func NewLayeredCheckResolver(
 	localResolverOpts []LocalCheckerOption,
 	cacheEnabled bool,
 	cachedResolverOpts []CachedCheckResolverOpt,
+	throttlingEnabled bool,
+	throttler throttler.Throttler,
+	dispatchThrottlingCheckConfig *DispatchThrottlingCheckResolverConfig,
 ) (CheckResolver, CheckResolverCloser) {
 	cycleDetectionCheckResolver := NewCycleDetectionCheckResolver()
 	localCheckResolver := NewLocalChecker(localResolverOpts...)
 
 	cycleDetectionCheckResolver.SetDelegate(localCheckResolver)
 
+	var dispatchThrottlingCheckResolver *DispatchThrottlingCheckResolver
```

Comment:
> If a cycle is dispatched it should return a cycle promptly
did you mean if a cycle is _found_?

i'm not sure i understood the reasoning behind the order but my gist is that the "cheap" layers should run first and let the more expensive ones run later

Reply:
I can do this but looks like we are doing in our server right now cc: @adriantam and it might need fixing

//	CycleDetectionCheckResolver  <-----|
//		DispatchThrottlingCheckResolver   |
//			CachedCheckResolver 			|
//				LocalChecker                   |
//					CycleDetectionCheckResolver -|